pkg/tsdb/azuremonitor/tokenprovider/authentication_middleware.go

```diff
@@ -3,26 +3,17 @@ package tokenprovider
 import (
```

Comment:
why not moving this with the rest of `aztokenprovider`? I will keep everything together since it's quite related.

And also, is there any change (apart from the `context` change) comparing with the existing `tokenprovider`?

Reply:
No changes besides the `ctx` for now.
Moved out definition of `TokenProvider` interface and `timeNow` variable as they are not related to middleware.